src/app/modules/main/profile_section/keycard/module.nim

```diff
@@ -157,52 +157,68 @@ method runCreatePukPopup*(self: Module) =
 method runCreateNewPairingCodePopup*(self: Module) =
   info "TODO: Create New Pairing Code for a Keycard..."
 
-proc buildKeycardList(self: Module) =
+proc buildKeycardItem(self: Module, walletAccounts: seq[WalletAccountDto], keyPair: KeyPairDto): KeycardItem =
   let findAccountByAccountAddress = proc(accounts: seq[WalletAccountDto], address: string): WalletAccountDto =
     for i in 0 ..< accounts.len:
       if(accounts[i].address == address):
         return accounts[i]
     return nil
 
-  let accounts = self.controller.getWalletAccounts()
+  var knownAccounts: seq[WalletAccountDto]
+  for accAddr in keyPair.accountsAddresses:
+    let account = findAccountByAccountAddress(walletAccounts, accAddr)
+    if account.isNil:
+      ## we should never be here cause we need to remove deleted accounts from the `keypairs` table and sync
```

Comment:
This looks familiar :)

Reply:
This is actually the same line as in my previous PR you commented on. :) Plan to work on this in a separate PR, where I need to handle wallet account deletion appropriately.